lib/ansible/modules/cloud/amazon/ec2_vpc_peer.py

```diff
@@ -416,11 +419,20 @@ def main():
         module.fail_json(msg="Can't authorize connection - " + str(e))
 
     if state == 'present':
+        if not vpc_id and not peer_vpc_id:
```

Comment:
you can use 

```
required_if([
    ('state', 'present', ['vpc_id', 'peer_vpc_id']),
    ('state', 'accept', ['peering_id']),
    ('state', 'reject', ['peering_id']),
 ])
```
Unfortunately the absent state check is too complicated for required_if

See https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/aws_direct_connect_gateway.py#L339-L340 for an example of using required_if

Reply:
Cool. Didn't know that existed. Also changed the absent check msg to be similar to what gets returned from required_if.